src/net/http/server.clj

```diff
@@ -146,40 +144,48 @@
       (channelInactive [^ChannelHandlerContext ctx]
         (close-channel ctx (:chan @state) true))
       (channelRead [^ChannelHandlerContext ctx msg]
-        (cond
-          (instance? HttpRequest msg)
-          (let [request (http/->request msg)]
-            (cond 
-              (bad? request)
-              (notify-bad-request! handler msg ctx (:chan @state)
-                                   "Trailing content on request")
-              
-              (= :net.http/unparsable-request request)
-              (notify-bad-request! handler msg ctx (:chan @state)
-                                   "Unparsable request")
-              
-              :else
-              (let [in      (a/chan inbuf)
-                    version (http/protocol-version msg)
-                    bodyreq (assoc request
-                                   :body in
-                                   :is-100-continue-expected? (HttpUtil/is100ContinueExpected msg)
-                                   :send-100-continue! (send-100-continue-fn ctx msg))]
-                (get-response (vswap! state assoc
-                                      :version version
-                                      :chan in
-                                      :request bodyreq)
-                              handler ctx executor))))
-
-          (chunk/content-chunk? msg)
-          (chunk/enqueue (:chan @state) ctx msg)
-
-          :else
-          (do
-            (buf/release msg)
-            (chan/close! ctx)
-            (a/close! (:chan @state))
-            (throw (IllegalArgumentException. "unhandled message chunk on body channel"))))))))
+        (let [chan (:chan @state)
+              close-resources (fn []
+                                (when (buf/buffer? msg)
+                                  (buf/release msg))
+                                (chan/close! ctx)
+                                (when chan
+                                  (a/close! chan)))]
+          (cond
+            (instance? HttpRequest msg)
+            (let [request (http/->request msg)
+                  version (http/protocol-version msg)
+                  respond! (partial write-response ctx version)]
+
+              (cond
+                (= :net.http/unparsable-request request)
+                (respond! (notify-bad-request! handler msg ctx nil "Unparsable request"))
```

Comment:
We are now writing the response to the client.
Previously it was not sent, only calling the handler which returns a ring HTTP response.